### PR TITLE
IGNITE-15346 Iterator of the ClientCacheQueryCursor can be closed during serialization.

### PR DESCRIPTION
**Pull request by NSAmelchev**
- Fix error message of closed iterator.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            ClientCache<Object, Object> cache = client.getOrCreateCache(DEFAULT_CACHE_NAME);
+
+            for (int i = 0; i < 10; i++)
+                cache.put(i, i);
+
+            ScanQuery<Object, Object> qry = new ScanQuery<>().setFilter((o, o2) -> {
+                try {
+                    filterLatch.countDown();
+
+                    U.sleep(1000);
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java:74]: Let's wait for another latch here (CountDown it after killQueries), in the current implementation we are waiting for each key (on each node) and this brings additional delay to test (about 5 keys per node - 5 seconds delay)

File: modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    /** Test logger. */
+    private ListeningTestLogger testLog = new ListeningTestLogger(log);
+
+    /** Query filter latch. */
+    private static CountDownLatch filterLatch = new CountDownLatch(1);
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testScanQueryClose() throws Exception {
+        try (Ignite srv1 = Ignition.start(Config.getServerConfiguration().setGridLogger(testLog));
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java:60]: Let's use test frameworks `startGrids(2)` with `getConfiguration` overriden to start server nodes (it works faster) and use `AbstractThinClientTest.startClient` to start thin client.

File: modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    /** Kill all queries. */
+    private static void killQueries() {
+        List<Ignite> srvs = G.allGrids();
+
+        assertFalse(srvs.isEmpty());
+
+        QueryMXBean bean = getMxBean(srvs.get(0).name(), "Query",
+            QueryMXBeanImpl.class.getSimpleName(), QueryMXBean.class);
+
+        for (IgniteEx ignite : F.transform(G.allGrids(), ignite -> (IgniteEx)ignite)) {
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java:110]: Perhaps we can connect to only one server node by thin client to simplify the test (kill all queries only on node we are connected to)

File: modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+import static org.apache.ignite.internal.managers.systemview.ScanQuerySystemView.SCAN_QRY_SYS_VIEW;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrowsAnyCause;
+
+/**
+ * Tests scan query close.
+ */
+public class ScanQueryCloseTest extends GridCommonAbstractTest {
+    /** Test logger. */
+    private ListeningTestLogger testLog = new ListeningTestLogger(log);
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/client/ScanQueryCloseTest.java:52]: final

File: modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java [hunk 5]
```diff
@@ -128,6 +154,68 @@ public static void doTestScanQueryCancel(IgniteEx cli, List<IgniteEx> srvs, Cons
...
+        assertTrue(filterLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+
+        List<List<?>> scanQries0 = execute(srvs.get(0),
+            "SELECT ORIGIN_NODE_ID, CACHE_NAME, QUERY_ID FROM SYS.SCAN_QUERIES");
+
+        assertEquals(1, scanQries0.size());
+
+        UUID originNodeId = (UUID)scanQries0.get(0).get(0);
+        String cacheName = (String)scanQries0.get(0).get(1);
+        long qryId = (Long)scanQries0.get(0).get(2);
```
[the remark this answers is not in the document]
Comment (NSAmelchev) [reply modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java]: Fixed.

File: modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java [hunk 6]
```diff
@@ -82,14 +87,35 @@
...
+
     /**
      * Test cancel of the scan query.
      *
      * @param cli Client node.
      * @param srvs Server nodes.
      * @param qryCanceler Query cancel closure.
      */
-    public static void doTestScanQueryCancel(IgniteEx cli, List<IgniteEx> srvs, Consumer<T3<UUID, String, Long>> qryCanceler) {
+    public static void doTestScanQueryCancel(IgniteEx cli, List<IgniteEx> srvs,
```
[the remark this answers is not in the document]
Comment (NSAmelchev) [reply modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java]: Fixed.

File: modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java [hunk 7]
```diff
@@ -82,14 +87,35 @@
     /** Latch to block compute task execution. */
     private static CountDownLatch computeLatch;
 
+    /** Scan query filter latch. */
+    private static CountDownLatch filterLatch;
```
[the remark this answers is not in the document]
Comment (NSAmelchev) [reply modules/indexing/src/test/java/org/apache/ignite/util/KillCommandsTests.java]: Fixed, thanks.

**Review by alex-plekhanov — APPROVED** [review]

**Pull request merged**